### ISSUE TITLE
0.24.0-beta-4 incorrectly handles plurals with `"` character; displays as `\"` on iOS and doesn't display on Android.

### ISSUE THREAD

**Issue by darronschall** [icerockdev/moko-resources#710]
Given this plural:

```xml
<plural name="search__results_for">
    <item quantity="zero">%d results for "%s"</item>
    <item quantity="one">%d result for "%s"</item>
    <item quantity="two">%d results for "%s"</item>
    <item quantity="few">%d results for "%s"</item>
    <item quantity="many">%d results for "%s"</item>
    <item quantity="other">%d results for "%s"</item>
</plural>
```

### iOS

The iOS plural generation in `Localizable.stringsdict` generates as:

```xml
<key>NSStringFormatSpecTypeKey</key>
<string>NSStringPluralRuleType</string>
<key>NSStringFormatValueTypeKey</key>
<string>d</string>
<key>zero</key>
<string>%%d results for \"%%s\"</string>
<key>one</key>
<string>%%d result for \"%%s\"</string>
<key>two</key>
<string>%%d results for \"%%s\"</string>
<key>few</key>
<string>%%d results for \"%%s\"</string>
<key>many</key>
<string>%%d results for \"%%s\"</string>
<key>other</key>
<string>%%d results for \"%%s\"</string>
```

The quotes in those `<string>` elements should _not_ be escaped. This is causing the string to show `\"` instead of just `"`:

<img width="150" alt="Screenshot 2024-05-16 at 9 48 42 AM" src="https://github.com/icerockdev/moko-resources/assets/129269/2335bdf3-ef53-4c63-8807-16ba34e8c3f6">

Note that replacing the `"` with `&quot;` does not make a difference - the generated output is still `\"`.

### Android

For Android, the generated content in `multiplatform_plurals.xml` _looks_ correct:

```xml
<plurals name="search__results_for">
    <item quantity="zero">%d results for &quot;%s&quot;</item>
    <item quantity="one">%d result for &quot;%s&quot;</item>
    <item quantity="two">%d results for &quot;%s&quot;</item>
    <item quantity="few">%d results for &quot;%s&quot;</item>
    <item quantity="many">%d results for &quot;%s&quot;</item>
    <item quantity="other">%d results for &quot;%s&quot;</item>
</plurals>
```

However, when using `MR.plurals.search__results_for.format(0, 0, "Example")`, the resulting string drops the quotes entirely:

<img width="150" alt="Screenshot 2024-05-16 at 10 22 20 AM" src="https://github.com/icerockdev/moko-resources/assets/129269/04717d14-d27a-4a31-9533-c3064652598a">

Is this an Android issue somewhere inside `PluralFormattedStringDesc`?

**Comment by ExNDY** [comment]
Hm, i'm not sure in android plurals, because i'm try two variants in native:

```
<resources>
    <plurals name="android_res_test">
        <item quantity="zero">zero chars &quot;%s&quot;</item>
        <item quantity="one">%d char &quot;%s&quot;</item>
        <item quantity="two">%d chars &quot;%s&quot;</item>
        <item quantity="other">%d chars &quot;%s&quot;</item>
    </plurals>
</resources>
```
And:
```
<resources>
    <plurals name="android_res_test">
        <item quantity="zero">zero chars "%s"</item>
        <item quantity="one">%d char "%s"</item>
        <item quantity="two">%d chars "%s"</item>
        <item quantity="other">%d chars "%s"</item>
    </plurals>
</resources>
```

Call inside activity:
```
this.resources.getQuantityString(R.plurals.android_res_test, 2, 2,"Same")
```
For all variants i'm get string without quotes. 
If i'm add mirror for quotes: like: `\"%s\"` or `\&quot;%s\&quot;` then i'm get quoting around Same.
And if i'm add mirror in plurals of moko-resource it will be fine for android, for ios i'm start research

**Comment by ExNDY** [comment]
will be fixed in 0.24.0-beta-5
